services/mgmt/resources/src/lib.rs

```diff
@@ -175,12 +175,14 @@ pub struct OperationConfig {
     pub api_version: String,
     pub client: reqwest::Client,
     pub base_path: String,
-    pub bearer_access_token: Option<String>,
+    pub bearer_access_token: Option<String>, // TODO remove
+    pub token_credential: Option<Box<dyn azure_core::TokenCredential>>,
+    pub token_credential_scope: String,
```

Comment:
Shouldn't it be a list of scopes?
https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-acquire-cache-tokens#scopes-when-acquiring-tokens
For v2.0, it is still in the `scope` http query param, but it is a space separated list. Let's looks and see what other SDKs do.

Reply:
I added an issue to review this in the not too distant future: https://github.com/Azure/azure-sdk-for-rust/issues/61